### PR TITLE
Add spec to the result of Coverage lib for `begin`

### PR DESCRIPTION
**Pull request by okuramasafumi**
Ref: https://github.com/jruby/jruby/issues/8173

I'm not sure if this is the right way to add a spec or not. Please point it out if I miss something or there's a better way.

### CODE REVIEW

File: library/coverage/code_with_begin_spec.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
+require_relative '../../spec_helper'
```
Comment (eregon) [inline library/coverage/code_with_begin_spec.rb:1]: Files almost always reflect a method, so could you move this example to https://github.com/ruby/spec/blob/master/library/coverage/result_spec.rb ?

Comment (okuramasafumi) [reply library/coverage/code_with_begin_spec.rb]: Sorry for being late, it's done.

File: library/coverage/result_spec.rb [hunk 2]
```diff
@@ -16,6 +17,7 @@
     $LOADED_FEATURES.delete(@class_file)
     $LOADED_FEATURES.delete(@config_file)
     $LOADED_FEATURES.delete(@eval_code_file)
+    $LOADED_FEATURES.delete(@file)
```
Comment (andrykonchin) [inline library/coverage/result_spec.rb:20]: I suppose it should use `@with_begin_file` instead of `@file`.

Comment (okuramasafumi) [reply library/coverage/result_spec.rb:20]: Yes, my mistake😅

**Comment by andrykonchin** [comment]
Thank you!

**Pull request merged**